src/components/Expander/Expander.tsx

```diff
@@ -108,32 +108,62 @@ const IfTitleTag = ({
   </Fragment>
 );
 
-export class ExpanderItem extends Component<ExpanderProps> {
+class ExpanderItem extends Component<ExpanderProps> {
   state: ExpanderState = {
     openState:
       this.props.defaultOpen !== undefined ? this.props.defaultOpen : false,
   };
 
-  handleClick = () => {
-    const {
-      open,
-      onClick,
-      consumer: { onClick: consumerOnClick } = { onClick: undefined },
-      index,
-      expanderGroup,
-    } = this.props;
-    if (!!expanderGroup && !!consumerOnClick && index !== undefined) {
-      consumerOnClick(index);
-    } else {
-      const openState = !this.state.openState;
-      const notControlled = open === undefined;
-      if (notControlled) {
-        this.setState({ openState });
+  componentDidUpdate(prevProps: ExpanderProps, prevState: ExpanderState) {
+    if (
+      !!this.props.consumer &&
+      !!prevProps.consumer &&
+      this.props.consumer.toggleAllExpanderState !==
+        prevProps.consumer.toggleAllExpanderState
+    ) {
+      const { openState } = this.state;
+      const {
+        expanderGroup,
+        index,
+        consumer: { toggleAllExpanderState },
+        open,
+      } = this.props;
+
+      if (
+        !!expanderGroup &&
+        index !== undefined &&
+        ((open === undefined &&
+          !!openState !== toggleAllExpanderState.toState) ||
+          (open !== undefined && !!open !== toggleAllExpanderState.toState))
+      ) {
+        this.handleClick();
       }
-      if (!!onClick) {
-        onClick({ openState: notControlled ? openState : !!open });
+    }
+
+    if (
+      prevState.openState !== this.state.openState ||
+      prevProps.open !== this.props.open
+    ) {
+      const {
+        expanderGroup,
+        index,
+        consumer: { onClick: consumerOnClick } = { onClick: undefined },
+      } = this.props;
+      if (!!expanderGroup && !!consumerOnClick && index !== undefined) {
+        consumerOnClick(index);
       }
     }
+  }
+
+  handleClick = () => {
+    const { open, onClick } = this.props;
+    const openState = !this.state.openState;
```

Comment:
this could maybe be written so that openState is resolved once based on presence of open prop and used on both setState and onClick.

Reply:
This is now simplified and `openState` is resolved just once and used for both.